### PR TITLE
feat: support scoped rules

### LINKED ISSUE #35: Support scoped packages
We are using GitHub Registry which currently only supports scoped packages. As a result, none of our packages can start with `bpmnlint-plugin-`. Is it possible to allow packages that are scoped, but start with that rule? An example would be `@myorg/bpmlint-plugin-my-org-rules`.

### PR DESCRIPTION
**Pull request by patoncrispy**
Intended to support scoped packages as highlighted in Issue https://github.com/bpmn-io/bpmnlint/issues/35.

Closes #35

### CODE REVIEW

**Comment by nikku** [comment]
For complete namespace support you'd need to update [`parseConfigName`](https://github.com/bpmn-io/bpmnlint/blob/master/lib/linter.js#L307) to support scoped packages, too.

**Comment by nikku** [comment]
Added failing integration test to your feature branch: https://github.com/bpmn-io/bpmnlint/pull/36/commits/2a6770a3c6dc3642f346ba03a02c04ed1357f928.

**Comment by patoncrispy** [comment]
Nice thanks @nikku will get on that now.

**Comment by patoncrispy** [comment]
Hey @nikku I keep getting errors running tests relating to `install-local`: `Error: Command failed with exit code 1 (EPERM): install-local`. That's all I have, unfortunately, but I'm not sure why I'm getting a permissions issue. The tests are failing in the before all hook of the cli-spec file.

File: lib/linter.js [hunk 1]
```diff
@@ -327,6 +327,22 @@ Linter.prototype.parseConfigName = function(name) {
     };
   }
 
+  if (name.toLowerCase().startsWith('plugin:@')) {
```
Comment (nikku) [inline lib/linter.js]: Please add a low level spec / unit test to verify the correct parsing behavior.

File: test/integration/cli/ns/extends.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (nikku) [inline test/integration/cli/ns/extends.json]: Please undo this change. 

The test is meant to verify that rules pulled in via `plugin:@ns/bpmnlint-plugin-test/recommended` are used without later activation (in the "rules" block).

**Comment by nikku** [comment]
> Hey @nikku I keep getting errors running tests relating to install-local: Error: Command failed with exit code 1 (EPERM): install-local. That's all I have, unfortunately, but I'm not sure why I'm getting a permissions issue. The tests are failing in the before all hook of the cli-spec file.

Please paste the full trace here.

File: test/integration/cli/ns/extends.json (hunk 2 above)
Comment (patoncrispy) [reply test/integration/cli/ns/extends.json]: But why is it using `test2` if we aren't extending from it? I may be misunderstanding something here...

**Comment by patoncrispy** [comment]
@nikku it's pretty useless, but here is what I have:
```
  1) cli
       "before all" hook in "cli":
     Error: Command failed with exit code 1 (EPERM): install-local
      at makeError (node_modules/execa/lib/error.js:58:11)
      at handlePromise (node_modules/execa/index.js:112:26)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

File: lib/linter.js (hunk 1 above)
Comment (patoncrispy) [reply lib/linter.js]: I've added the test.

File: test/integration/cli/ns/extends.json (hunk 2 above)
Comment (patoncrispy) [reply test/integration/cli/ns/extends.json]: OK got it.

**Comment by patoncrispy** [comment]
I extracted the `stderr` from `execa` and got the following:

```
Command failed: npm i --no-save "/var/folders/cg/vy26v0hd53j2dhxfsnstjw5w0000gn/T/node-local-install-sktwictuk1usuali/bpmnlint-6.2.0.tgz" "/var/folders/cg/vy26v0hd53j2dhxfsnstjw5w0000gn/T/node-local-install-sktwictuk1usuali/bpmnlint-plugin-test-0.0.0.tgz"
npm ERR! path /Users/me/code/bpmnlint/test/integration/cli/node_modules/bpmnlint
npm ERR! code EISGIT\nnpm ERR! git /Users/me/code/bpmnlint/test/integration/cli/node_modules/bpmnlint: Appears to be a git repo or submodule.
npm ERR! git     /Users/me/code/bpmnlint/test/integration/cli/node_modules/bpmnlint
npm ERR! git Refusing to remove it. Update manually,\nnpm ERR! git or move it out of the way first.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2019-10-17T14_28_16_315Z-debug.log

    at ChildProcess.exithandler (child_process.js:294:12)
    at ChildProcess.emit (events.js:198:13)
    at maybeClose (internal/child_process.js:982:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
  killed: false,
  code: 1,
  signal: null,
  cmd:
   \'npm i --no-save "/var/folders/cg/vy26v0hd53j2dhxfsnstjw5w0000gn/T/node-local-install-sktwictuk1usuali/bpmnlint-6.2.0.tgz" "/var/folders/cg/vy26v0hd53j2dhxfsnstjw5w0000gn/T/node-local-install-sktwictuk1usuali/bpmnlint-plugin-test-0.0.0.tgz"\' }',
```

**Comment by patoncrispy** [comment]
@nikku seems to pass those tests now - could you take a look?

File: lib/linter.js [hunk 3]
```diff
@@ -224,7 +222,7 @@ Linter.prototype.resolveConfiguredRules = function(config) {
 Linter.prototype.lint = function(moddleRoot, config) {
 
   config = config || this.config;
-
+  // throw new Error(JSON.stringify(config));
```
Comment (nikku) [inline lib/linter.js]: Please remove this line.

File: lib/linter.js [hunk 4]
```diff
@@ -76,9 +76,7 @@ Linter.prototype.applyRule = function applyRule(moddleRoot, ruleDefinition) {
 
 };
 
-
```
Comment (nikku) [inline lib/linter.js]: Please undo style formating.

File: lib/linter.js [hunk 5]
```diff
@@ -76,9 +76,7 @@ Linter.prototype.applyRule = function applyRule(moddleRoot, ruleDefinition) {
...
 Linter.prototype.resolveRule = function(name) {
-
```
Comment (nikku) [inline lib/linter.js]: Please undo style formating.

File: test/spec/linter-spec.js [hunk 6]
```diff
@@ -399,12 +409,13 @@ describe('linter', function() {
...
               'bpmnlint:recommended',
-              'plugin:test/recommended'
+              'plugin:test/recommended',
+              'plugin:@ns/bpmnlint-plugin-test2/recommended'
             ],
             rules: {
               'bpmnlint/foo': 'error',
               'foo': 'error',
-              'test/bar': 'off'
+              'test/bar': 'off',
```
Comment (nikku) [inline test/spec/linter-spec.js]: Please remove trailing `,`.

File: test/spec/linter-spec.js [hunk 7]
```diff
@@ -413,10 +424,12 @@ describe('linter', function() {
 
           // then
           expect(rules).to.eql({
+            '@ns/bpmnlint-plugin-test2/bar': 'off',
             'bar': 'error',
             'foo': 'error',
             'test/bar': 'off',
-            'test/other': 'warn'
+            'test/other': 'warn',
```
Comment (nikku) [inline test/spec/linter-spec.js]: Please drop `,` and newline.

Comment (patoncrispy) [reply test/spec/linter-spec.js]: Done.

File: test/spec/linter-spec.js (hunk 6 above)
Comment (patoncrispy) [reply test/spec/linter-spec.js]: Done.

File: lib/linter.js (hunk 5 above)
Comment (patoncrispy) [reply lib/linter.js]: Done.

File: lib/linter.js (hunk 4 above)
Comment (patoncrispy) [reply lib/linter.js]: Done.

File: lib/linter.js (hunk 3 above)
Comment (patoncrispy) [reply lib/linter.js]: Done.

**Comment by patoncrispy** [comment]
hey @nikku any chance you could take another look at this?

**Comment by nikku** [comment]
Giving it another :eyes: is on my TODO list. :+1:

**Comment by nikku** [comment]
@philippfromme Can be reviewed despite the CI failure. I'll ship a follow up PR to silence the Node@latest warning.

**Pull request merged**